hydra-plutus/src/Hydra/Contract/Commit.hs

```diff
@@ -68,16 +67,14 @@ serializeCommit (i, o) = do
 
 -- | Decode an on-chain 'SerializedTxOut' back into an off-chain 'TxOut'.
 -- NOTE: Depends on the 'Serialise' instance for Plutus' 'Data'.
-deserializeCommit :: Commit -> Maybe (OffChain.TxIn, OffChain.TxOut CtxUTxO)
-deserializeCommit Commit{input, preSerializedOutput} =
+deserializeCommit :: OffChain.NetworkId -> Commit -> Maybe (OffChain.TxIn, OffChain.TxOut CtxUTxO)
```

Comment:
@ch1bo Should I avoid this change so I keep the scripts the same as in `0.9.0` release?

Reply:
It's unavoidable, this is required to commit anything from Mainnet.

We can put it into another PR to make the change more deliberate though.

Reply:
I forgot there was another change to v_commit a part from this one. All good.

Reply:
Wait.. I was wrong.

This is not a breaking change in the scripts! This function is only used off-chain. While the interface of this function changes (breaking the API), the scripts should have the same script hash.

See: https://hydra.family/head-protocol/benchmarks/transaction-cost/#script-summary and https://github.com/input-output-hk/hydra/pull/715#issuecomment-1422754935

This means, that the 0.9.0 scripts can actually be used on mainnet already.